index.html

```diff
@@ -2795,8 +2826,7 @@ <h3><dfn>Writing or pushing content</dfn></h3>
                 </dl>
               </li>
               <li>
-                If |message| is of type {{NDEFMessageInit}} and its
-                |id:string| is not `null`:
+                If |record|'s |id| is not `null`:
```

Comment:
Also, for the `undefined` case, we might want to set `NDEFRecord#id` to the current origin (like https://example.com/) rather than doing nothing? Then later when we try to filter records with `NDEFScanOptions#id` being https://example.com/, this record can be found out.

Reply:
I'm actually not sure we want to do this.
@kenchris and @zolkis What do you think?
If web app doesn't specify any `id`, should it write `https://example.com/` in the record id?

My main concerns is that I'm not sure web developers would expect this behavior when they simply call `writer.push('foo')` as this would make it harder to guess the size of the NDEFMessage to push.

Reply:
Currently we don't any more use origin policy (Web NFC record and id), so I don't see any point in setting `id` to origin. It should be application's decision.

Reply:
By reading the example at https://w3c.github.io/web-nfc/#save-and-restore-game-progress-with-another-device, it implies that if `id` is specified as `/mypath/mygame/update`, we write it as `https://mygame.com/mypath/mygame`. But if `id` is not specified we just skip writing it.
The logic seems not that tight.

Reply:
And, the `urlRecord` naming in https://w3c.github.io/web-nfc/#creating-a-record-identifier looks like some leftover from the old era. I'm not sure, are we already done refining this algorithm and it's complying with the example code mentioned above?

Reply:
Yes, we should clean them out. 
The example can still contain an id, actually it is nice to have example with id, since now it is in application domain.

Reply:
@zolkis Can you share what the example should look like?

Reply:
id: document.location.origin ?

I am fine with us not setting this by default

Reply:
My point is that we should give a meaningful value to `NDEFRecord#id` if it's not specified by NDEFRecordInit, whether the value should be the origin or not does not matter.

Reply:
I'm not sure why `NDEFRecord#id` should have a meaningful value if web dev didn't specify one. @leonhsl Can you explain? Sorry for being slow.

Reply:
If we leave `NDEFRecord#id` undefined, it means we will not write any id information into a nfc tag for this record.
Then, later `scan({id: 'https://example.com'})` will NOT match it, which I think is **incorrect** behavior, because callers expect to see all records written by pages in https://example.com. 
While the record with `/mypath/myid` gets its id transformed to `https://example.com/mypath/myid` and will hit the match.

Reply:
Actually our id generation mechanism is already implying that the record id written by WebNFC API is bound inherently with the current web page, so I suggest to push through this rule everywhere.

We've already been doing some internal transformation over the ids (binding them to the current host), so I don't think generating `https://example.com` for an `undefined `id will take users by surprise.

Reply:
Right, there is a mismatch.
We have canceled the idea to write Web NFC id (first as separate id record, then as record id), but we did not clean that up completely. Now we need to decide what we want: do we keep it completely, or do we remove it completely, but now it's hanging between two ideas, indeed.

If we take removing this "additional smartness" consistently, then we should fix (or remove) the id generation algorithm. It should be something the application explicitly wants. We could include some convenience (like a boolean flag whether to include it), but apps can initialize `id` themselves, for instance it could be a public key or a registrable domain etc.

Reply:
Shall we open a separate issue for the Web NFC id?
I'd be in favor of removing this "id" constraint algorithm and let web developers do what ever they want with it.

```js
const writer = new NDEFWriter();
await writer.push({ records: [{ recordType: 'text', data: 'hi', id: '123' }] });
...
const reader = new NDEFReader();
await reader.scan({id: '123' });
reader.onreading = event => {};
```

Reply:
> I'd be in favor of removing this "id" constraint algorithm and let web developers do what ever they want with it.

I agree. Let's track this in an issue.

Reply:
Let's move forward at the issue https://github.com/w3c/web-nfc/issues/445.